src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -101,6 +134,16 @@ def rpc_method(method: Callable[..., _T]) -> Callable[..., _T]:
     return method
 
 
+def regional_rpc_method(
```

Comment:
I don't think that's a concern. The type annotations on `regional_rpc_method` don't actually have any effect on the runtime annotations of the method it returns. It passes through the original method body (after tampering with its attrs slightly), and when we call `inspect.signature` we're seeing the same object as the `method` arg below.

Also, an annotation like `Callable[[_CT], _Ct]` wouldn't work because that constrains the method to have a single argument (`[_CT]`). The `_CT` value sounds like the single model object that we use to serialize, but we don't do that until later.

Reply:
The second half isn't quite true -- the typing here is the decorator, so the single _CT is, the full type of the callable being wrapped.

I agree that likely, the annotations that are in python state remain the same passing through a decorator.  My question is -- does the mypy "view" of the "decorated function" change to the more coarse typing of `Callable[..., Any]` as a result of the decorator?